drivers/dma/dma_ti_mspm0.c

```diff
@@ -0,0 +1,344 @@
+		*increment = DL_DMA_ADDR_DECREMENT;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static inline int dma_ti_mspm0_get_dstdatawidth(uint8_t wd, uint32_t *dwidth)
+{
+	if (dwidth == NULL) {
+		return -EINVAL;
+	}
+
+	switch (wd) {
+	case DMA_TI_MSPM0_DATAWIDTH_BYTE:
+		*dwidth = DL_DMA_WIDTH_BYTE;
+		break;
+	case DMA_TI_MSPM0_DATAWIDTH_HALF:
+		*dwidth = DMA_DMACTL_DMADSTWDTH_HALF;
+		break;
+	case DMA_TI_MSPM0_DATAIWDTH_WORD:
+		*dwidth = DMA_DMACTL_DMADSTWDTH_WORD;
+		break;
+	case DMA_TI_MSPM0_DATAWIDTH_LONG:
+		*dwidth = DMA_DMACTL_DMADSTWDTH_LONG;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static inline int dma_ti_mspm0_get_srcdatawidth(uint8_t wd, uint32_t *dwidth)
+{
+	if (dwidth == NULL) {
+		return -EINVAL;
+	}
+
+	switch (wd) {
+	case DMA_TI_MSPM0_DATAWIDTH_BYTE:
+		*dwidth = DL_DMA_WIDTH_BYTE;
+		break;
+	case DMA_TI_MSPM0_DATAWIDTH_HALF:
+		*dwidth = DMA_DMACTL_DMASRCWDTH_HALF;
+		break;
+	case DMA_TI_MSPM0_DATAIWDTH_WORD:
+		*dwidth = DMA_DMACTL_DMASRCWDTH_WORD;
+		break;
+	case DMA_TI_MSPM0_DATAWIDTH_LONG:
+		*dwidth = DMA_DMACTL_DMASRCWDTH_LONG;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_ti_mspm0_configure(const struct device *dev, uint32_t channel,
+				  struct dma_config *config)
+{
+	uint32_t temp;
+	const struct dma_ti_mspm0_config *cfg = dev->config;
+	struct dma_ti_mspm0_data *dma_data = dev->data;
+	struct dma_ti_mspm0_channel_data *data = NULL;
+	struct dma_block_config *b_cfg = config->head_block;
+	DL_DMA_Config dma_cfg = {0};
+
+	if ((config == NULL) || (channel > DMA_TI_MSPM0_MAX_CHANNEL)) {
+		return -EINVAL;
+	}
+
+	data = &dma_data->ch_data[channel];
+
+	if (data->busy != false) {
+		return -EBUSY;
+	}
+
+	if (config->dest_data_size != config->source_data_size) {
+		LOG_ERR("Source and Destination data width is not same");
+		return -EINVAL;
+	}
+
+	if (dma_ti_mspm0_get_memory_increment(b_cfg->source_addr_adj, &temp)) {
+		LOG_ERR("Invalid Source address increment");
+		return -EINVAL;
+	}
+
+	dma_cfg.srcIncrement = temp;
+
+	if (dma_ti_mspm0_get_memory_increment(b_cfg->dest_addr_adj, &temp)) {
+		LOG_ERR("Invalid Destination address increment");
+		return -EINVAL;
+	}
+
+	dma_cfg.destIncrement = temp;
+
+	if (dma_ti_mspm0_get_dstdatawidth(config->dest_data_size, &temp)) {
+		LOG_ERR("Invalid Destination data width");
+		return -EINVAL;
+	}
+
+	dma_cfg.destWidth = temp;
+
+	if (dma_ti_mspm0_get_srcdatawidth(config->source_data_size, &temp)) {
+		LOG_ERR("Invalid Source data width");
+		return -EINVAL;
+	}
+
+	dma_cfg.srcWidth = temp;
+	data->direction = config->channel_direction;
+	data->dma_callback = config->dma_callback;
+	data->user_data = config->user_data;
+	dma_cfg.transferMode = DL_DMA_SINGLE_TRANSFER_MODE,
+	dma_cfg.extendedMode = DL_DMA_NORMAL_MODE,
+	dma_cfg.triggerType = DL_DMA_TRIGGER_TYPE_EXTERNAL;
+	dma_cfg.trigger	= config->dma_slot;
+
+	DL_DMA_clearInterruptStatus(DMA, channel + DMA_TI_MSPM0_BASE_CHANNEL_NUM);
```

Comment:
It's unclear here if there are any shared registers, but if any register is shared the register update needs to be done inside a spin lock (critical section)